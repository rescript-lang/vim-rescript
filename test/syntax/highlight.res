@@warning("-27")
//^

%%raw(`console.log("Hello")`)
//^

@deprecated
//^
let _ = %raw(`1`)
        //^

let a = 1
//^

let _ = 1
      //^

let _ = true
        //^
let _ = false
        //^

let _ = "hello!"
          //^
let _ = `Hello World ðŸ‘‹ ${name}`
                  //^
let _ = 'a'
       //^

let _ = j`Today`
          //^

let _ = %re("/b/g")
       //^
let _ = true && true
           //^

let _ = true || false
           //^

let _ = !true
      //^

let _ = 2 <= 3
        //^

let _ = 2 <= 3
        //^

type t
//^

type r
   //^

type t = {
  age: int
      //^
}

type t = {
  a: promise<int>
      //^
}

let _ = list{1, 2}
        //^

let _ = {...me, age: me.age + 1}
          //^

let _ = Some("My Name")
        //^

let _ = #red
        //^

let _ = if true {
      //^
  "Good morning!"
} else {
  //^
  "Hello!"
}

for i in 1 to 10 {
//^
  Js.log(i)
 //^
}

while true {
//^
  Js.log(true)
}

exception InputClosed(string)
//^


try {
//^
  someJSFunctionThatThrows()
} catch {
  //^
| Not_found => assert false
                //^
}

open Js.Array2
//^

include Js
//^


switch [1, 2, 3] {
//^
  | [] => true
//^
  | _ => false
    //^
}

let list = 2
    //^
let array = [1]
    //^
let catch = 1
    //^
let ref = ref(false)
   //^
let option = None
    //^

let _ = sql`select * from table`
        //^

type a = dict<string>
         //^

type b = result<int, string>
         //^

let _ = (x: r1) => (x :> r2)
                    //^

foo(1, ...)
      //^
