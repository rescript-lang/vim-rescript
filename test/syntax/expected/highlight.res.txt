{ {
    col = 1,
    row = 0,
    syntax = { {
        hl_group = "resAttribute",
        hl_group_link = "PreProc"
      } }
  }, {
    col = 1,
    row = 3,
    syntax = { {
        hl_group = "resExtension",
        hl_group_link = "PreProc"
      } }
  }, {
    col = 1,
    row = 6,
    syntax = { {
        hl_group = "resAttribute",
        hl_group_link = "PreProc"
      } }
  }, {
    col = 9,
    row = 8,
    syntax = { {
        hl_group = "resExtension",
        hl_group_link = "PreProc"
      } }
  }, {
    col = 1,
    row = 11,
    syntax = { {
        hl_group = "resKeyword",
        hl_group_link = "Statement"
      } }
  }, {
    col = 7,
    row = 14,
    syntax = {}
  }, {
    col = 9,
    row = 17,
    syntax = { {
        hl_group = "resBoolean",
        hl_group_link = "Constant"
      } }
  }, {
    col = 9,
    row = 19,
    syntax = { {
        hl_group = "resBoolean",
        hl_group_link = "Constant"
      } }
  }, {
    col = 11,
    row = 22,
    syntax = { {
        hl_group = "resString",
        hl_group_link = "String"
      } }
  }, {
    col = 19,
    row = 24,
    syntax = { {
        hl_group = "resString",
        hl_group_link = "String"
      } }
  }, {
    col = 8,
    row = 26,
    syntax = { {
        hl_group = "resChar",
        hl_group_link = "Constant"
      } }
  }, {
    col = 11,
    row = 29,
    syntax = { {
        hl_group = "resString",
        hl_group_link = "String"
      } }
  }, {
    col = 8,
    row = 32,
    syntax = { {
        hl_group = "resExtension",
        hl_group_link = "PreProc"
      } }
  }, {
    col = 12,
    row = 34,
    syntax = {}
  }, {
    col = 12,
    row = 37,
    syntax = {}
  }, {
    col = 7,
    row = 40,
    syntax = {}
  }, {
    col = 9,
    row = 43,
    syntax = {}
  }, {
    col = 9,
    row = 46,
    syntax = {}
  }, {
    col = 1,
    row = 49,
    syntax = { {
        hl_group = "resKeyword",
        hl_group_link = "Statement"
      } }
  }, {
    col = 4,
    row = 52,
    syntax = {}
  }, {
    col = 7,
    row = 56,
    syntax = { {
        hl_group = "resNone",
        hl_group_link = "resNone"
      }, {
        hl_group = "resType",
        hl_group_link = "Type"
      } }
  }, {
    col = 7,
    row = 61,
    syntax = { {
        hl_group = "resNone",
        hl_group_link = "resNone"
      }, {
        hl_group = "resType",
        hl_group_link = "Type"
      } }
  }, {
    col = 9,
    row = 65,
    syntax = { {
        hl_group = "resFunction",
        hl_group_link = "Function"
      } }
  }, {
    col = 11,
    row = 68,
    syntax = { {
        hl_group = "resNone",
        hl_group_link = "resNone"
      } }
  }, {
    col = 9,
    row = 71,
    syntax = { {
        hl_group = "resModuleOrVariant",
        hl_group_link = "Function"
      } }
  }, {
    col = 9,
    row = 74,
    syntax = { {
        hl_group = "resPolyVariant",
        hl_group_link = "Function"
      } }
  }, {
    col = 7,
    row = 77,
    syntax = {}
  }, {
    col = 3,
    row = 80,
    syntax = { {
        hl_group = "resKeyword",
        hl_group_link = "Statement"
      } }
  }, {
    col = 1,
    row = 85,
    syntax = { {
        hl_group = "resRepeat",
        hl_group_link = "Statement"
      } }
  }, {
    col = 2,
    row = 87,
    syntax = { {
        hl_group = "resNone",
        hl_group_link = "resNone"
      }, {
        hl_group = "resModuleChain",
        hl_group_link = "PreProc"
      } }
  }, {
    col = 1,
    row = 91,
    syntax = { {
        hl_group = "resRepeat",
        hl_group_link = "Statement"
      } }
  }, {
    col = 1,
    row = 96,
    syntax = { {
        hl_group = "resException",
        hl_group_link = "Statement"
      } }
  }, {
    col = 1,
    row = 100,
    syntax = { {
        hl_group = "resException",
        hl_group_link = "Statement"
      } }
  }, {
    col = 3,
    row = 103,
    syntax = { {
        hl_group = "resException",
        hl_group_link = "Statement"
      } }
  }, {
    col = 17,
    row = 105,
    syntax = { {
        hl_group = "resNone",
        hl_group_link = "resNone"
      }, {
        hl_group = "resException",
        hl_group_link = "Statement"
      } }
  }, {
    col = 1,
    row = 109,
    syntax = { {
        hl_group = "resInclude",
        hl_group_link = "PreProc"
      } }
  }, {
    col = 1,
    row = 112,
    syntax = { {
        hl_group = "resInclude",
        hl_group_link = "PreProc"
      } }
  }, {
    col = 1,
    row = 116,
    syntax = { {
        hl_group = "resKeyword",
        hl_group_link = "Statement"
      } }
  }, {
    col = 1,
    row = 118,
    syntax = { {
        hl_group = "resNone",
        hl_group_link = "resNone"
      } }
  }, {
    col = 5,
    row = 120,
    syntax = { {
        hl_group = "resNone",
        hl_group_link = "resNone"
      } }
  }, {
    col = 5,
    row = 124,
    syntax = { {
        hl_group = "resType",
        hl_group_link = "Type"
      } }
  }, {
    col = 5,
    row = 126,
    syntax = { {
        hl_group = "resType",
        hl_group_link = "Type"
      } }
  }, {
    col = 5,
    row = 128,
    syntax = { {
        hl_group = "resException",
        hl_group_link = "Statement"
      } }
  }, {
    col = 4,
    row = 130,
    syntax = { {
        hl_group = "resType",
        hl_group_link = "Type"
      } }
  }, {
    col = 5,
    row = 132,
    syntax = { {
        hl_group = "resType",
        hl_group_link = "Type"
      } }
  } }